api-metastore/src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -1,33 +1,86 @@
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.EventTypeBase;
 import org.zalando.nakadi.domain.EventTypeSchema;
+import org.zalando.nakadi.domain.EventTypeSchemaBase;
 import org.zalando.nakadi.domain.PaginationWrapper;
+import org.zalando.nakadi.exceptions.runtime.InvalidEventTypeException;
 import org.zalando.nakadi.exceptions.runtime.InvalidLimitException;
 import org.zalando.nakadi.exceptions.runtime.InvalidVersionNumberException;
 import org.zalando.nakadi.exceptions.runtime.NoSuchSchemaException;
+import org.zalando.nakadi.plugin.api.authz.AuthorizationService;
+import org.zalando.nakadi.repository.db.EventTypeRepository;
 import org.zalando.nakadi.repository.db.SchemaRepository;
+import org.zalando.nakadi.util.JsonUtils;
+import org.zalando.nakadi.validation.JsonSchemaEnrichment;
+import org.zalando.nakadi.validation.SchemaIncompatibility;
 
+import java.io.InputStream;
+import java.util.List;
+import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 @Component
 public class SchemaService {
 
     private static final Pattern VERSION_PATTERN = Pattern.compile("\\d+\\.\\d+\\.\\d+");
-    private static final Logger LOG = LoggerFactory.getLogger(SchemaService.class);
 
     private final SchemaRepository schemaRepository;
     private final PaginationService paginationService;
+    private final JsonSchemaEnrichment jsonSchemaEnrichment;
+    private final SchemaEvolutionService schemaEvolutionService;
+    private final EventTypeRepository eventTypeRepository;
+    private final AdminService adminService;
+    private final AuthorizationValidator authorizationValidator;
+    private final EventTypeCache eventTypeCache;
 
     @Autowired
     public SchemaService(final SchemaRepository schemaRepository,
-                         final PaginationService paginationService) {
+                         final PaginationService paginationService,
+                         final JsonSchemaEnrichment jsonSchemaEnrichment,
+                         final SchemaEvolutionService schemaEvolutionService,
+                         final EventTypeRepository eventTypeRepository,
+                         final AdminService adminService,
+                         final AuthorizationValidator authorizationValidator,
+                         final EventTypeCache eventTypeCache) {
         this.schemaRepository = schemaRepository;
         this.paginationService = paginationService;
+        this.jsonSchemaEnrichment = jsonSchemaEnrichment;
+        this.schemaEvolutionService = schemaEvolutionService;
+        this.eventTypeRepository = eventTypeRepository;
+        this.adminService = adminService;
+        this.authorizationValidator = authorizationValidator;
+        this.eventTypeCache = eventTypeCache;
+    }
+
+    public void addSchema(final EventType originalEventType, final EventTypeSchemaBase newSchema) {
```

Comment:
This is new code (well, sort of since I just follow the same pattern used in other services, with respect to authorization and how cache is used).